src/test/java/com/google/devtools/build/lib/skylark/SkylarkDefinedAspectsTest.java

```diff
@@ -904,6 +904,39 @@ public void aspectFailingOrphanArtifacts() throws Exception {
             + "test/missing_in_action.txt\n");
   }
 
+  @Test
+  public void aspectSkippingOrphanArtifactsWithLocation() throws Exception {
+    scratch.file("aspects/BUILD", "");
+    scratch.file(
+        "aspects/print.bzl",
+        "def _print_expanded_copts_impl(target, ctx):",
+        "    # Make sure the rule has a srcs attribute.",
+        "    if hasattr(ctx.rule.attr, 'cmd'):",
```

Comment:
nit: when is this false?

Reply:
This test case was simplified from a more complicated one... it's leftover from the old one. Same for all the other comments... I've reduced the test case a bunch.